src/jsonpatcherproxy.js

```diff
@@ -181,6 +226,30 @@ var JSONPatcherProxy = (function() {
     }
     return this.generateProxyAtPath(root, path);
   };
+  JSONPatcherProxy.prototype._resetCachedProxiesPaths = function(root, path) {
+    /* deleting array elements could render other array elements paths incorrect, 
+    this function fixes all incorrect paths efficiently */
+    for (let key in root) {
+      if (
```

Comment:
maybe it would be efficient to store a reference to `root[key]` here not to access it 5 times bellow